backend/game/cards/keep_cards/energy_manipulation_cards/energy_hoarder.py

```diff
@@ -0,0 +1,16 @@
+from game.cards.card import Card
+from game.cards.keep_card import KeepCard
+
+
+class EnergyHoarder(KeepCard):
+    def __init__(self):
+        Card.__init__(self, "Energy Hoarder", 3,
+                      "You gain 1[Star] for every 6[Energy] you have at the end of your turn.", "")
+
+    def immediate_effect(self, player_that_bought_the_card, other_players):
+        card = EnergyHoarder()
+        player_that_bought_the_card.add_card(card)
```

Comment:
Is there no way to bring this up to KeepCard?

Reply:
immediate effect add_card is only applicable for keep cards, for discard cards the cards do not need to be added to card store

Reply:
That did not answer the question in any way.

Reply:
at this level i can implement the abstract class defined in keep card for each card, adding them individually easier is what i meant

Reply:
"easier" is not ok.
Every single keepcard is going to have the exact same immediate effect - add it to the player.
We need to avoid this code duplication if at all possible.

Reply:
gone